apps/src/javalab/JavalabSettings.jsx

```diff
@@ -0,0 +1,118 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import Radium from 'radium';
+import color from '@cdo/apps/util/color';
+import onClickOutside from 'react-onclickoutside';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+
+const styles = {
+  main: {
+    display: 'inline-block'
+  },
+  dropdown: {
+    border: `1px solid ${color.charcoal}`,
+    position: 'absolute'
+  },
+  anchor: {
+    padding: 10,
+    color: color.charcoal,
+    backgroundColor: color.white,
+    fontFamily: '"Gotham 5r", sans-serif',
+    display: 'block',
+    textDecoration: 'none',
+    lineHeight: '20px',
+    transition: 'background-color .2s ease-out',
+    ':hover': {
+      backgroundColor: color.lightest_gray,
+      cursor: 'pointer'
+    }
+  },
+  nonFirstAnchor: {
+    borderTop: `1px solid ${color.charcoal}`
+  }
+};
+
+/**
+ * A button that drops down to a set of clickable links, and closes itself if
+ * you click on the button, or outside of the dropdown.
+ */
+export const JavalabSettings = class JavalabSettingsComponent extends Component {
+  static propTypes = {
+    style: PropTypes.object,
+    children: props => {
+      React.Children.map(props.children, child => {
+        if (child.type !== 'a') {
+          throw new Error('only accepts children of type <a/>');
+        }
+        if (!child.props.onClick) {
+          throw new Error('each child must have an href or onclick');
+        }
+      });
+    }
+  };
+
+  state = {
+    dropdownOpen: false
+  };
+
+  expandDropdown = () => {
```

Comment:
This component would benefit from some tests coverage of the various state changes to future proof as we add more items to settings.